### PR TITLE
feat: cleanup task executions older than x days

### PR DESCRIPTION
**Pull request by klopfdreh**
Needs to be integrated with: https://github.com/spring-cloud/spring-cloud-dataflow/pull/5422

Bildschirmfoto 2023-08-02 um 11 06 42

You can enter a number of days and if the task executions are older than before 5 days for example they are removed.

See #1949

### CODE REVIEW

**Comment by oodamien** [comment]
Hi @klopfdreh,
The use case is interesting. 
We have to work a bit more on the UX/UI.

**Comment by klopfdreh** [comment]
Hey @oodamien - no problem, the backend implementation is working for us and also fixes a bug with boot 3 / batch 5. The underscore is misaligned with the link so it would be great to see how you would implement it. 👍

**Comment by onobc** [comment]
Just for a status update on the backend... we are very close to getting it merged into SCDF main.

**Comment by oodamien** [comment]
Update: Working on the UI/UX, I should push changes soon.

**Comment by klopfdreh** [comment]
Thanks for the update and for your work - I am very interested in how you changed the UI implementation @oodamien

**Comment by onobc** [comment]
The backend changes have been merged in https://github.com/spring-cloud/spring-cloud-dataflow/commit/af60a25693e9c7d302cdc9c6fcd2e6297cbb0a34

**Comment by klopfdreh** [comment]
For the changes:

"taskJob": "Task/Job Executions"
"taskJob": "Aufgaben-Ausführung"
"taskJob": "выполнение задач/заданий"

"content": "You can filters the task/job execution(s) you want to clean up:"
"content": "Sie können die Task-/Job-Ausführungen filtern, die Sie aufräumen wollen:"
"content": "Вы можете отфильтровать выполнение задач/заданий, которые хотите очистить:"

"onlyComplete": "Only complete execution(s)"
"onlyComplete": "Nur beendete Ausführung(en)"
"onlyComplete": "Только полное выполнение(я)"

"olderExecution": "Cleanup older execution(s)"
"olderExecution": "Aufräumen älterer Ausführung(en)"
"olderExecution": "Очистка старых исполнений"

"days": "Days"
"days": "Tage"
"days": "Дни"

"daysInvalid": "Invalid days"
"daysInvalid": "Tage ungültig"
"daysInvalid": "Неверные дни"

"successContent": "Execution(s) cleaned up."
"successContent": "Ausführung(en) erfolgreich aufgeräumt."
"successContent": "Исполнение(я) очищено."

German is valid 😄 - Russian should be checked.

The CSS `tools.component.css` can be removed as the markup is not used anymore.

**Comment by klopfdreh** [comment]
Can you provide a screenshot so that I can see the appearance, now? 😃

I am currently busy by solving some issues at work, so I don’t know when I am able to check it out and test.

**Comment by oodamien** [comment]
Sure :

image

image

image

**Comment by onobc** [comment]
Looks good @oodamien ,

One thing to point out is that "Only complete execution(s)" does not make sense when "Cleanup older execution(s)" as the "end_date" is used as the date anchor. IOW - the executions must be completed in order to determine how many days old they are.

**Comment by oodamien** [comment]
I see. Let's make the "Cleanup older execution(s)" visible only if complete is checked.

image

image

image

**Comment by klopfdreh** [comment]
Awesome! 👍

**Review by claudiahub — APPROVED** [review]
LGTM, about Russian translations I've double checked on en to tu translators

File: ui/src/app/manage/tools/tools.component.html [hunk 1]
```diff
@@ -22,20 +22,39 @@ <h1>{{ 'tools.title' | translate }}</h1>
...
   <div class="clr-col-lg-12 clr-col-md-12 clr-col-12" appRole appFeature="tasks">
     <app-view-card
-      [titleModal]="'tools.cleanUp' | translate"
+      [titleModal]="'tools.taskJob' | translate"
       keyContext="import-export"
       name="import-export"
       id="cleanup"
     >
       <ng-template>
+        <!-- <div class="cleanUpGrid"> -->
```
Comment (klopfdreh) [inline ui/src/app/manage/tools/tools.component.html:31]: Comment can be removed.

File: ui/src/app/manage/tools/tools.component.html [hunk 2]
```diff
@@ -22,20 +22,39 @@ <h1>{{ 'tools.title' | translate }}</h1>
...
-          <a href="{{ baseApiUrl }}#" (click)="run('cleanup-all')">{{ 'tools.cleanUpAll' | translate }}</a>
-        </div>
-        <div>
-          <clr-icon shape="angle" style="transform: rotate(90deg)"></clr-icon>
-          <a href="{{ baseApiUrl }}#" (click)="run('cleanup-completed')">{{ 'tools.cleanUpCompleted' | translate }}</a>
+          <div>
+            <clr-icon shape="angle" style="transform: rotate(90deg)"></clr-icon>
+            <a href="{{ baseApiUrl }}#" (click)="run('cleanup-all')">{{ 'tools.cleanUpAll' | translate }}</a>
+          </div>
+          <!-- <div>
```
Comment (klopfdreh) [inline ui/src/app/manage/tools/tools.component.html:50]: Comment can be removed.

File: ui/src/app/manage/tools/cleanup/cleanup.component.spec.ts [hunk 3]
```diff
@@ -48,18 +48,18 @@ describe('manage/tools/cleanup/cleanup.component.ts', () => {
...
-    const title = fixture.debugElement.query(By.css('.modal-title-wrapper')).nativeElement;
-    expect(title.textContent).toContain('Confirm Clean Up Task Execution(s)');
-    fixture.debugElement.query(By.css('.modal-footer .btn-danger')).nativeElement.click();
-    fixture.detectChanges();
-    await fixture.whenStable();
-    expect(NotificationServiceMock.mock.successNotifications[0].title).toBe('Clean up execution(s)');
-    expect(NotificationServiceMock.mock.successNotifications[0].message).toBe('12 execution(s) cleaned up.');
-    done();
-  });
+  // it('should clean up the task executions', async done => {
```
Comment (klopfdreh) [inline ui/src/app/manage/tools/cleanup/cleanup.component.spec.ts:51]: Is this test going to be reenabled?

**Comment by oodamien** [comment]
Thanks @klopfdreh

**Pull request merged**